Sources/MongoSwift/ChangeStream.swift

```diff
@@ -0,0 +1,165 @@
+import mongoc
+
+/// A token used for manually resuming a change stream. Pass this to the `resumeAfter` or `startAfter` fields of
+/// `ChangeStreamOptions` to resume or start a change stream where a previous one left off.
+/// - SeeAlso: https://docs.mongodb.com/manual/changeStreams/#resume-a-change-stream
+public struct ChangeStreamToken: Codable {
+    private let resumeToken: Document
+
+    public init(resumeToken: Document) {
```

Comment:
If that's the case, the `resumeToken` still gets updated/extracted from the `_id` field in `next`. @kmahar I think this init should not be public then.

Reply:
@patrickfreed  If that is the case, wouldn't the resume token still get updated from the `id` field? @kmahar  brought up whether this init needed to be public or not. So maybe this init should be internal?

Reply:
@patrickfreed Actually, when I first added the `ChangeStreamToken` struct, it only had this public init. Then we added the decoder init when we encountered this decoding error: `"No value associated with key CodingKeys(stringValue: \"resumeToken\", intValue: nil) (\"resumeToken\").", underlyingError: nil))`
When the `resumeToken` gets updated from the `_id` field in the document returned from libmongoc, it always uses the decoder init. 
I think I am not seeing or understanding how this public init will be used/needed now. If the user opted for `ChangeStream<Document>` wouldn't the `ChangeStream.resumeToken` still get updated in `next` and just extract it from there?

Reply:
>I think I am not seeing or understanding how this public init will be used/needed now. If the user opted for ChangeStream<Document> wouldn't the ChangeStream.resumeToken still get updated in next and just extract it from there?

You're absolutely right, good point! I forgot that we already cache it for them. In that case, this initializer doesn't need to be public. In fact, since you said we only use the decoder initializer, it doesn't actually have to exist at all, right?

Reply:
@patrickfreed Actually, both inits are needed since `ChangeStream.resumeToken` is a non-optional and needs to be initialized in `ChangStream.init`. It also used when updating the `resumeToken` in `next`.
I think this init can just be `internal`?

Reply:
Oh, right. Yep `internal` is good.